File: Source/Website/ViewModels/Base/ReactViewModel.cs
using Website.Dto;

namespace Website.ViewModels.Base
{
    /// <summary>
    /// ViewModel to be passed to React client- and server-rendering code.
    /// </summary>
    public class ReactViewModel
    {
        public string Page { get; set; }
        public string BaseUrl { get; set; }
        public IanaTimeZone TimeZone { get; set; }

        public ViewModel ViewModel { get; set; }
        public bool IsDebug { get; set; }
    }
}

File: Source/Website/Scripts/Models/ReactViewModel.ts



import { IanaTimeZone } from './IanaTimeZone';
import { ExampleUserDto } from './ExampleUserDto';
import { ErrorDto } from './ErrorDto';
import { ViewModel } from './ViewModel';


export const ReactViewModelTypeName = 'ReactViewModel'
export interface ReactViewModel  { 
	page: string
	baseUrl: string
	viewModel: ViewModel
}
